sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -636,6 +648,20 @@ private static void verifyUnsplittableMethods(ErrorReporter errors, DoFnSignatur
               outputT));
     }
 
+    for (Parameter param : extraParameters) {
+      errors.checkArgument(
+          Predicates.or(
+                  Predicates.instanceOf(Parameter.OnTimerContextParameter.class),
```

Comment:
Make a constant set and for each check that at least one of `isAssignableFrom()`?